src/ambianic/pipeline/store.py

```diff
@@ -171,18 +175,52 @@ def process_sample(self, **sample) -> Iterable[dict]:
                 yield processed_sample
 
     def notify(self, save_json: dict):
-        if self.notification is None:
-            return
-        log.debug("Sending notification(s)..")
         # TODO extract inference data
+        if save_json['inference_result'] is None:
+            return
+
         for inference_result in save_json['inference_result']:
             data = {
                 'id': save_json['id'],
                 'label': inference_result['label'],
                 'confidence': inference_result['confidence'],
                 'datetime': save_json['datetime'],
             }
-            notification = Notification(data=data, providers=self.notification_config["providers"])
+
+            try:
+                premiumFile = pkg_resources.resource_filename(
+                    "ambianic.webapp", "premium.yaml")
+                with open(premiumFile, "r+") as file:
+                    configFile = yaml.safe_load(file)
+                    userId = configFile['credentials']["USER_AUTH0_ID"]
+                    endpoint = configFile['credentials']["NOTIFICATION_ENDPOINT"]
+                    if (userId):
```

Comment:
This section with premium notification logic should be in the `ambianic.notification` module , not hardcoded in `store.py`.

Reply:
Alright. This adjustment has been made. 

I created a function ( isolated from the `Notification` class ) in the `Notification` file to handle sending the emails using the Cloud API.